### DISCUSSION TITLE
`optional` with `default` is marked nullable in `Output` type

### DISCUSSION THREAD

**Discussion by ItsMeBrianD** [open-circle/valibot#473] [Q&A]
Doing my first exploration with Valibot this morning, and running into something that feels a little counterintuitive.

Given some schema:

```javascript
const myObjectType = object({
  arrayField: optional(array(string()), ['Hi!']),
});

/*
Resulting type:
{ arrayField?: string[] }

Expected Result:
{ arrayField: string[] }
*/

type MyObjectType = Output<typeof myObjectType>;

const ThisShouldFail: MyObjectType = {};
```

Because `MyObjectType` is defined using `Output`, I would expect `arrayField` to be required, rather than being marked optional.
Is this due to something I am doing incorrectly?

**Comment by fabian-hiller** [comment]
It works as it should for me. You can have a look [here](https://valibot.dev/playground/?code=JYWwDg9gTgLgBAKjgQwM5wG5wGZQiOAcg2QBtgAjCGQgbgCh6BjCAO1XhAE8B5CgKwCmTGABUuYQXAC8mAHQQBwmAAoA3vTgooUZFwBiwQaQAmALnkQwMYGzIqMc5Dr0O5HKMFYBzFQEo-ABo4AG1CAAlgAEJCAF0g+gBfPwZ6GAkpAFleJRFxSRl5HgBXGDBSgB50yQhsOG4+ITyMgD5UlnZ4UQALYFQAZW6IYtN9ZGBSC2zG5XypWTVE2iA) in the playground.
